aws-logs-loggroup/src/main/java/software/amazon/logs/loggroup/UpdateHandler.java

```diff
@@ -28,6 +30,15 @@
             putRetentionPolicy(proxy, request, logger);
         }
 
+        // It can take up to five minutes for the (dis)associate operation to take effect
+        // It's unclear from the documentation if that state can be checked via the API.
+        // https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/encrypt-log-data-kms.html
```

Comment:
Are we emitting ProgressEvents with `IN_PROGRESS` to check on the status or it seems like that's not actually discoverable?

Reply:
Where does the 5 mins come from?  As far as I see there is no such status in the log group. It may take some time for data plane to take effect, but from control plane perspective, it's done once API call successful.

Reply:
The docs say "After you associate or disassociate a CMK from a log group, it can take up to five minutes for the operation to take effect." To me that sounds like the data plane needing some time, like Wenbing described,and what I assumed when writing the code. There is no separate API call to check on the status and it's hard to  test because "it is immediately  present when I do a read call", still falls into the "up to 5 minutes" bucket.

Reply:
The docs say "After you associate or disassociate a CMK from a log group, it can take up to five minutes for the operation to take effect." To me that sounds like the data plane needing some time, like Wenbing described,and what I assumed when writing the code. There is no separate API call to check on the status and it's hard to test because "it is immediately present when I do a read call", still falls into the "up to 5 minutes" bucket.